qualtran/bloqs/block_encoding/sparse_matrix.py

```diff
@@ -239,14 +250,31 @@ def build_composite_bloq(
 
 
 @frozen
-class FullRowColumnOracle(RowColumnOracle):
-    """Oracle specifying the non-zero rows or columns of a matrix with full entries."""
+class TopLeftRowColumnOracle(RowColumnOracle):
+    """Oracle specifying the non-zero rows or columns of a matrix with top left block non-zero.
+
+    Args:
+        system_bitsize: The number of bits used to represent an index.
+        num_nonzero: The number of rows or columns of the non-zero top left block. If unspecified,
+            defaults to making the entire matrix non-zero.
+
+    Registers:
+        l: As input, index specifying the `l`-th non-zero entry to find in row / column `i`.
+           As output, position of the `l`-th non-zero entry in row / column `i`.
+        i: The row / column index.
+    """
 
     system_bitsize: SymbolicInt
+    _num_nonzero: SymbolicInt = field()
+
+    @_num_nonzero.default
+    def _num_nonzero_default(self):
+        # Default to a matrix with all entries non-zero
+        return 2**self.system_bitsize
 
     @cached_property
     def num_nonzero(self) -> SymbolicInt:
-        return 2**self.system_bitsize
+        return self._num_nonzero
```

Comment:
I think you can just use `num_nonzero` as an attribute. As the class is frozen, it can't be modified.

Reply:
That's true in principle, but pylint doesn't accept having `num_nonzero` be an attribute here in order to override the `@property @abstractmethod` named `num_nonzero` in the parent class:
```
qualtran/bloqs/block_encoding/sparse_matrix_test.py:85:17: E0110: Abstract class 'TopLeftRowColumnOracle' with abstract methods instantiated (abstract-class-instantiated)
```

Hence, I had to redundantly declare the property with a different name from the attribute.

Reply:
Ah didn't realize it was an abstract property in the parent class